text/0249-v2-experiments.md

```diff
@@ -364,17 +244,167 @@ Disadvantages:
 
 ### Force a naming convention for unstable code
 
-We can modify JSII to force a certain naming convention for unstable code, for
-example to add a specific prefix or suffix to all unstable APIs.
+We can modify `awslint` to force a certain naming convention for unstable code,
+for example to add a specific prefix or suffix to all unstable APIs.
 
 Advantages:
 
 - Should fulfill goal #1 - it will be impossible to use an unstable API by
   accident.
+- Does not require changes in JSII, only in `awslint`.
 
 Disadvantages:
 
 - Will force some pretty long and ugly names on our APIs.
 - Graduating a module will require a lot of code changes from our customers to
   remove the prefix/suffix.
-- Requires changes in JSII.
+
+# Appendix A - why can't we break backwards compatibility in the code of mono-CDK main modules?
+
+This section explains why it will not be possible to break backwards compatibility of any API inside the stable modules of mono-CDK.
+
+Imagine we could break backwards compatibility in the code of the  `aws-cdk-lib` main modules.
+The following scenario would then be possible:
+
+Let's say we have a third-party library, `my-library`, that vends `MyConstruct`.
+It's considered stable by its author.
+However, inside the implementation of `MyConstruct`,
+it uses an experimental construct, `SomeExperiment`, from mono-CDK's S3 module.
+It's just an implementation detail, though; it's not reflected in the API of `MyConstruct`.
+
+`my-library` is released in version `2.0.0`,
+and it has a peer dependency on `aws-cdk-lib` version `2.10.0`
+(with a caret, so `"aws-cdk-lib": "^2.10.0"`).
+
+Some time passes, enough that `aws-cdk-lib` is now in version `2.20.0`.
+A CDK customer wants to use `my-library` together with the newest and shiniest `aws-cdk-lib`,`2.20.0`,
+as they need some recently released features.
+However, incidentally, in version `2.15.0` of `aws-cdk-lib`,
+`SomeExperiment` was broken -- which is fine, it's an experimental API.
+Suddenly, the combination of `my-library` `2.0.0` and `aws-cdk-lib` `2.20.0`
+will fail for the customer at runtime,
+and there's basically no way for them to unblock themselves other than pinning to version `2.14.0` of `aws-cdk-lib`,
+which was exactly the problem mono-CDK was designed to prevent in the first place.
+
+# Appendix B - modules depending on unstable modules
+
+This section contains the snapshot of the interesting dependencies between Construct Library modules as of writing this document.
+
+## Stable modules depending on unstable modules
+
+```
+⚠️  Stable module '@aws-cdk/aws-applicationautoscaling' depends on unstable module '@aws-cdk/aws-autoscaling-common'
```

Comment:
I know this has already been merged, but for the next revision...

I think it would be really helpful to identify 1-3 categories of these dependencies, and then have a very quick statement about how Option N would handle this.

Potential categories:
* Implementation detail -- the module relies on an experimental construct as an implementation detail, completely hidden from the user.
* Targets/"Join modules" -- The dependency exists to bridge between two services (e.g., stepfunctions-tasks and batch). As new services are added, new targets will need to be created.
* Exposed as props/`add*` methods - E.g., Lambda/EFS

How would option 2/5 tackle each of the above?